QModManager/API/SMLHelper/Handlers/CraftTreeHandler.cs

```diff
@@ -0,0 +1,338 @@
+        /// <param name="displayName">The display name of the tab, which will show up when you hover your mouse on the tab.</param>
+        /// <param name="sprite">The sprite of the tab.</param>
+        /// <param name="stepsToTab">
+        /// <para>The steps to the target tab.</para>
+        /// <para>These must match the id value of the CraftNode in the crafting tree you're targeting.</para>
+        /// <para>Do not include "root" in this path.</para>
+        /// </param>        
+        void ICraftTreeHandler.AddTabNode(CraftTree.Type craftTree, string name, string displayName, UnityEngine.Sprite sprite, params string[] stepsToTab)
+        {
+            ValidateStandardCraftTree(craftTree);            
+            string modName = ReflectionHelper.CallingAssemblyByStackTrace().GetName().Name;
+
+            CraftTreePatcher.TabNodes.Add(new TabNode(stepsToTab, craftTree, new Atlas.Sprite(sprite), modName, name, displayName));
+        }
+
+        /// <summary>
+        /// <para>Removes a node at the specified node location. Can be used to remove either tabs or craft nodes.</para>
+        /// <para>If a tab node is selected, all child nodes to it will also be removed.</para>
+        /// </summary>
+        /// <param name="craftTree">The target craft tree to edit.</param>
+        /// <param name="stepsToNode">
+        /// <para>The steps to the target node.</para>
+        /// <para>These must match the id value of the CraftNode in the crafting tree you're targeting.</para>
+        /// <para>This means matching the id of the crafted item or the id of the tab name.</para>
+        /// <para>Do not include "root" in this path.</para>
+        /// </param>
+        void ICraftTreeHandler.RemoveNode(CraftTree.Type craftTree, params string[] stepsToNode)
+        {
+            ValidateStandardCraftTree(craftTree);
+            CraftTreePatcher.NodesToRemove.Add(new Node(stepsToNode, craftTree));
+        }
+
+        #endregion
+
+        private static void ValidateStandardCraftTree(CraftTree.Type craftTree)
+        {
+            switch (craftTree)
+            {
+                case CraftTree.Type.Fabricator:
+                case CraftTree.Type.Constructor:
+                case CraftTree.Type.Workbench:
+                case CraftTree.Type.SeamothUpgrades:
+                case CraftTree.Type.MapRoom:
+                case CraftTree.Type.Centrifuge:
+                case CraftTree.Type.CyclopsFabricator:
+                case CraftTree.Type.Rocket:
+                    break; // Okay
+                case CraftTree.Type.Unused1:                    
+                case CraftTree.Type.Unused2:                    
+                case CraftTree.Type.None:
+                default:
+                    throw new ArgumentException($"{nameof(craftTree)} value of '{craftTree}' does not correspond to a standard crafting tree. This method is intended for use only with standard crafting trees, not custom ones or unused ones.");
```

Comment:
Can we do this with `<`/`>`?

Reply:
The invalid cases are `int` values 0, 4, and 5.
```CSharp
public enum Type
{
    None = 0,
    Fabricator = 1,
    Constructor = 2,
    Workbench = 3,
    Unused1 = 4,
    Unused2 = 5,
    SeamothUpgrades = 6,
    MapRoom = 7,
    Centrifuge = 8,
    CyclopsFabricator = 9,
    Rocket = 10
}
```
Normally, the compiler would crush a switch statement like this into a simple math check.  
But in this case, the values to check for aren't in a continuous range.  
So there's no easy math that can be done.

The compiler has done the best that can be done already which is this:
![image](https://user-images.githubusercontent.com/39146191/61185303-b14b1780-a625-11e9-8f68-2ed301a1bd4d.png)